flyingpigeon/utils.py

```diff
@@ -1110,3 +1114,83 @@ def swap_free(self):
     @property
     def swap_used(self):
         return self._convert * self._swapu
+
+
+def guess_main_variable(ncdataset):
```

Comment:
After verification, it seems that ocgis' rd.variable (used in utils.get_variable) returns only variables that have all of 'time', 'lon', 'lat' as dimensions. 2d (lat, lon) fields, 3d (level, lat, lon) fields are not returned. This is rather arbitrary (my function is also arbitrary as it returns only the highest dimensionality variable). What do we want to support here? Obvisouly the variables we subset/average must have lon/lat, then I see the following possibilities : (time, notime), (level, nolevel), multivariables (including possibility of lower dimensionality fields, i.e. both a field with tzyx and another with only yx in the same NetCDF). The more we support the more tests we have to write...

Reply:
My guess is that whatever we want to support also makes sense for ocgis. My suggestion would be to leave this as is for now but plan to make improvements to ocgis to support our use cases that are not currently supported.  For example, pick the variables with tzxy if it exists, otherwise pick txy, otherwise zxy, ortherwise xy.